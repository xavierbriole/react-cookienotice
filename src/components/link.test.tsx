import { render } from '@testing-library/react'

import Link from './link'

describe('Link', () => {
  describe('should render', () => {
    it('with all props', () => {
      const { asFragment } = render(<Link to='#' newTab={true} label='label' />)

      expect(asFragment()).toMatchSnapshot()
    })

    it('without props', () => {
      const { asFragment } = render(
        <Link to={undefined} newTab={undefined} label={undefined} />,
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })

  it('should display link', () => {
    const { getByText } = render(<Link to='#' newTab={true} label='label' />)

    expect(getByText('label')).toBeInTheDocument()
  })

  it('should open in new tab', () => {
    const { getByText } = render(<Link to='#' newTab={true} label='label' />)

    const link = getByText('label')

    expect(link.getAttribute('target')).toBe('_blank')
    expect(link.getAttribute('rel')).toBe('noreferrer')
  })

  it('should open in same tab', () => {
    const { getByText } = render(<Link to='#' newTab={false} label='label' />)

    const link = getByText('label')

    expect(link.getAttribute('target')).toBe('_self')
    expect(link.getAttribute('rel')).toBe(null)
  })
})
