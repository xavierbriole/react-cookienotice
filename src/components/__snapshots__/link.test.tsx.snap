// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > should render 1`] = `
<a
  class="react-cookienotice-link"
  href="#"
  rel="noreferrer"
  target="_blank"
>
  label
</a>
`;
